pandas/core/accessor.py

```diff
@@ -129,3 +102,136 @@ def f(self, *args, **kwargs):
             # don't overwrite existing methods/properties
             if overwrite or not hasattr(cls, name):
                 setattr(cls, name, f)
+
+
+# Ported with modifications from xarray
+# https://github.com/pydata/xarray/blob/master/xarray/core/extensions.py
+# 1. We don't need to catch and re-raise AttributeErrors as RuntimeErrors
+
+
+class _CachedAccssor(object):
```

Comment:
name this: ``CachedAccessor`` we use this internally so private is not appropriate